walletlibrary/src/main/java/com/microsoft/walletlibrary/util/VerifiedIdResult.kt

```diff
@@ -1,3 +1,58 @@
 package com.microsoft.walletlibrary.util
 
-typealias VerifiedIdResult<T> = Result<T>
\ No newline at end of file
+import com.microsoft.did.sdk.util.controlflow.NetworkException
+import com.microsoft.did.sdk.util.controlflow.SdkException
+
+typealias VerifiedIdResult<T> = Result<T>
+
+internal suspend fun <T> getResult(block: suspend () -> T): VerifiedIdResult<T> {
+    return try {
+        val result = block()
+        VerifiedIdResult.success(result)
+    } catch (verifiedIdException: VerifiedIdException) {
+        verifiedIdException.toVerifiedIdResult()
+    } catch (exception: WalletLibraryException) {
+        when (exception.cause) {
+            is NetworkException -> {
+                val networkException = exception.cause as NetworkException
+                val networkingException = NetworkingException(
+                    exception.message ?: "",
+                    VerifiedIdExceptions.NETWORKING_EXCEPTION.value,
+                    networkException.correlationVector,
+                    networkException.errorCode,
+                    networkException.innerErrorCodes,
+                    networkException.errorBody,
+                    networkException.retryable
+                )
+                networkingException.cause = networkException
+                networkingException.toVerifiedIdResult()
+            }
+            is SdkException -> {
```

Comment:
Will every SdkException be Malformed? What is an SdkException?

Reply:
Any Exception that happens in VC SDK is SDKException. If it involves network calls, then it is NetworkException (subtype of SDKException) hence caught first. This exception happens when there is any issue during validation or request/response processing which are all malformed in a way.